hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -355,6 +466,14 @@ func MachineDeploymentComplete(deployment *capiv1.MachineDeployment) bool {
 		newStatus.ObservedGeneration >= deployment.Generation
 }
 
+func DeploymentComplete(deployment *appsv1.Deployment) bool {
```

Comment:
I think we should be checking the deployment conditions instead of these granular checks — it's up to the k8s deployment controller to expose these higher level concepts in terms of the lower level API fields. For example, check Available=True and Progressing=False.

Consider for example that deployment availability is based on things like the minimum availability spec on the deployment, so direct equality comparisons like you have here are only valid assuming certain fixed minimum availability values.

Reply:
We can fix this in a followup